docs/pages/guides/react-compiler.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: React Compiler
+description: Learn how to enable and use the React Compiler in Expo apps.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+import { Step } from '~/ui/components/Step';
+
+> Warning: React Compiler is still experimental!
+
+The new [React Compiler](https://react.dev/learn/react-compiler) automatically memoizes components and hooks to enable fine-grained reactivity. This can lead to significant performance improvements in your app. The React Compiler is currently experimental and is not enabled by default. You can enable it in your app by following the instructions below.
+
+## Enabling React Compiler
+
+First, [check how compatible](https://react.dev/learn/react-compiler#checking-compatibility) your project is with the React Compiler.
+
+<Terminal cmd={['$ npx react-compiler-healthcheck@latest']} />
+
+This will generally verify if your app is following the [**rules of React**](https://react.dev/reference/rules).
+
+Next, toggle on the React Compiler experiment in your `app.json` file:
+
+```json app.json
+{
+  "experiments": {
+    "reactCompiler": true
+  }
+}
+```
```

Comment:
````suggestion
<Step label="1">

[Check how compatible](https://react.dev/learn/react-compiler#checking-compatibility) your project is with the React Compiler.

<Terminal cmd={['$ npx react-compiler-healthcheck@latest']} />

This will generally verify if your app is following the [**rules of React**](https://react.dev/reference/rules).

</Step>

<Step label="2">

Toggle on the React Compiler experiment in your app config file:

```json app.json
{
  "experiments": {
    "reactCompiler": true
  }
}
```

</Step>
````

- Let's use `Step` component here to list the steps.
- Use filenames as bold as per our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text).
- Update the verbiage slightly after adding steps (no need for using words like first or next).